src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -12013,13 +11939,149 @@ public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y)
             public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y, Vector512<T> z) => (x * y) + z;
         }
 
+        /// <summary>(x * y) + z</summary>
+        internal readonly struct MultiplyAddEstimateOperator<T> : ITernaryOperator<T> where T : INumberBase<T>
+        {
+            public static T Invoke(T x, T y, T z)
+            {
+                // TODO https://github.com/dotnet/runtime/issues/98053: Use T.MultiplyAddEstimate when it's available.
+
+                if (typeof(T) == typeof(Half))
```

Comment:
We want all 3 floating-point scalar paths here to use `if (Fma.IsSupported || AdvSimd.IsSupported)` before calling `FusedMultiplyAdd`.

Basically, if those are `false` then the `T.FusedMultiplyAdd` function ends up executing code that looks like this: https://github.com/amd/aocl-libm-ose/blob/master/src/isa/avx/fmaf.c

This is quite a bit slower than `(x * y) + z` (but more accurate), so we don't want to use it unless we kknow its accelerated and will emit the dedicated ~4 cycle instruction instead.